src/Mod/Part/App/FCRepAlgoAPI_BooleanOperation.cpp

```diff
@@ -0,0 +1,56 @@
+/***************************************************************************
+ *   Copyright (c) 2024 Eric Price (CorvusCorax) <eric.price@tuebingen.mpg.de>
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+/**
+  * FCRepAlgoAPI provides a wrapper for various OCCT functions.
+  */
+
+#include <FCRepAlgoAPI_Section.h>
+#include <BRepBndLib.hxx>
+#include <Bnd_Box.hxx>
+#include <TopoDS_Shape.hxx>
+#include <Precision.hxx>
+#include <App/Application.h>
+
+FCRepAlgoAPI_BooleanOperation::FCRepAlgoAPI_BooleanOperation()
+:
+  BRepAlgoAPI_BooleanOperation()
+{
+    SetFuzzyValue(FCRepAlgoAPI_BooleanOperation::getDefaultFuzzyValue(0.0));
+}
+
+
+FCRepAlgoAPI_BooleanOperation::FCRepAlgoAPI_BooleanOperation(const TopoDS_Shape& theS1,
+                                               const TopoDS_Shape& theS2,
+                                               const BOPAlgo_Operation theOperation)
+: BRepAlgoAPI_BooleanOperation(theS1, theS2, theOperation)
+{
+    Bnd_Box bounds;
+    BRepBndLib::Add(theS1, bounds);
+    BRepBndLib::Add(theS2, bounds);
+    SetFuzzyValue(FCRepAlgoAPI_BooleanOperation::getDefaultFuzzyValue(bounds.SquareExtent()));
+}
+  
+double FCRepAlgoAPI_BooleanOperation::getDefaultFuzzyValue(const double size) {
+    Base::Reference<ParameterGrp> hGrp = App::GetApplication().GetUserParameter()
```

Comment:
Please do not add a direct dependency to FreeCAD's Application and ParameterGrp classes. Instead add a helper class similar to OCC's Precision to get and set the fuzzy value.

The value should be initialized inside `PyMOD_INIT_FUNC(Part)` and set inside `DlgSettingsGeneral::saveSettings()`

Reply:
can you show me an example? I just looked in other Part/App/... files how user parameters are read and this was how.

Note that FCRepAlgoAPI_BooleanOperation::getDefaultFuzzyValue() is a static method, it doesn't have access to instantiated properties of the class (since I could not make FCRep_Algo_API_Section a child class of FCRepAlgoAPI_BooleanOperation because of double-inheritance conflicts)

Reply:
to be honest, I am not too familiar with the FreeCAD code and how things are supposed to be done, I only started playing with it 2 weeks ago.

When I tried to do this, I looked for examples in the same model. I used this example as a template on how to access settings:

https://github.com/FreeCAD/FreeCAD/blob/ad54b21d54d075c00d3836bbeaaa602ec06ff189/src/Mod/Part/App/FeaturePartBoolean.cpp#L121-L132

if this is wrong, I really need an example on how to do it correctly

Reply:
in DlgSettingsGeneral::saveSettings()
this only says
ui->checkBooleanCheck->onSave();
so I copied that for the FuyyBoolean entry and also used onSave() - I have no clue what it does though, must be QWidget specific

Reply:
I moved getDefaultFuzzyValue into its own helper class now - but I totally did not understand the part about PyMOD_INIT_FUNC(Part)

Reply:
ok i think i got it now. ill try

Reply:
and done